process/transaction/interceptedTransaction_test.go

```diff
@@ -809,6 +864,118 @@ func TestInterceptedTransaction_CheckValidityOkValsShouldWork(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestInterceptedTransaction_CheckValiditySignedWithHashButNotEnabled(t *testing.T) {
+	t.Parallel()
+
+	minTxVersion := uint32(1)
+	chainID := []byte("chain")
+	tx := &dataTransaction.Transaction{
+		Nonce:     1,
+		Value:     big.NewInt(2),
+		Data:      []byte("data"),
+		GasLimit:  3,
+		GasPrice:  4,
+		RcvAddr:   recvAddress,
+		SndAddr:   senderAddress,
+		Signature: sigOk,
+		ChainID:   chainID,
+		Version:   minTxVersion + 1,
+		Options:   16777216,
+	}
+
+	marshalizer := &mock.MarshalizerMock{}
+	txBuff, _ := marshalizer.Marshal(tx)
+	shardCoordinator := mock.NewMultipleShardsCoordinatorMock()
+	shardCoordinator.CurrentShard = 6
+	shardCoordinator.ComputeIdCalled = func(address []byte) uint32 {
+		if bytes.Equal(address, senderAddress) {
+			return senderShard
+		}
+		if bytes.Equal(address, recvAddress) {
+			return recvShard
+		}
+
+		return shardCoordinator.CurrentShard
+	}
+
+	txi, _ := transaction.NewInterceptedTransaction(
+		txBuff,
+		marshalizer,
+		marshalizer,
+		mock.HasherMock{},
+		createKeyGenMock(),
+		createDummySigner(),
+		&mock.PubkeyConverterStub{},
+		shardCoordinator,
+		createFreeTxFeeHandler(),
+		&mock.WhiteListHandlerStub{},
+		&mock.ArgumentParserMock{},
+		chainID,
+		minTxVersion,
+		false,
+		mock.HasherMock{},
+	)
+
+	err := txi.CheckValidity()
+	assert.Equal(t, process.ErrTransactionSignedWithHashIsNotEnabled, err)
+}
+
+func TestInterceptedTransaction_CheckValiditySignedWithHashShoudWork(t *testing.T) {
+	t.Parallel()
+
+	minTxVersion := uint32(1)
+	chainID := []byte("chain")
+	tx := &dataTransaction.Transaction{
+		Nonce:     1,
+		Value:     big.NewInt(2),
+		Data:      []byte("data"),
+		GasLimit:  3,
+		GasPrice:  4,
+		RcvAddr:   recvAddress,
+		SndAddr:   senderAddress,
+		Signature: sigOk,
+		ChainID:   chainID,
+		Version:   minTxVersion + 1,
+		Options:   16777216,
```

Comment:
maybe move that number to core-constants and name it properly

Reply:
refactored.